tests/test_hydraulic_power.py

```diff
@@ -238,6 +238,164 @@ def test_hydraulic_power(self):
+
+            def energy_system_options(self):
+                options = super().energy_system_options()
+
+                self.gas_network_settings[
+                    "head_loss_option"] = HeadLossOption.LINEARIZED_N_LINES_EQUALITY
+                self.gas_network_settings["n_linearization_lines"] = 3
+                self.gas_network_settings["minimum_velocity"] = 0.0
+                self.gas_network_settings["minimize_head_losses"] = True
+
+                return options
+
+        solution = run_optimization_problem(
+            GasProblemHydraulic,
+            base_folder=base_folder,
+            esdl_file_name="test.esdl",
+            esdl_parser=ESDLFileParser,
+            profile_reader=ProfileReaderFromFile,
+            input_timeseries_file="timeseries.csv",
+        )
+
+        # TODO: add check on values for hydraulic power.
+        results = solution.extract_results()
+
+        pipes = ["Pipe_7c53", "Pipe_f1a4", "Pipe_0e39", "Pipe_c50f"]
+        for pipe in pipes:
+            pipe_hp_in = results[f"{pipe}.GasIn.Hydraulic_power"]
+            pipe_hp_out = results[f"{pipe}.GasOut.Hydraulic_power"]
+            pipe_hp = results[f"{pipe}.Hydraulic_power"]
+
+            pipe_mass = results[f"{pipe}.GasIn.mass_flow"]
+            pipe_vol_flow = results[f"{pipe}.GasOut.Q"]
+
+            np.testing.assert_allclose(pipe_hp, pipe_hp_in - pipe_hp_out)
+            if pipe in ["Pipe_7c53", "Pipe_c50f"]:
+                # connected to demand, thus hydraulic_power should be 0
+                np.testing.assert_allclose(0, pipe_hp_out)
+
+
+            v_max = solution.gas_network_settings["maximum_velocity"]
+
+            v_inspect = results[f"{pipe}.GasOut.Q"] / solution.parameters(0)[f"{pipe}.area"]
+            v_points = [i * v_max / solution.gas_network_settings["n_linearization_lines"] for i in
+                        range(solution.gas_network_settings["n_linearization_lines"] + 1)]
+
+            # due to non linearity, every timestep on new linearized line, a doubled mass flow should result in more than doubled hydraulic power
+            v_inspect_line_ind = []
+            for k in range(len(v_inspect)):
+                for i in range(len(v_points)):
+                    if v_points[i]+1e-6< v_inspect[k] <=v_points[i+1]+1e-6:
+                        v_inspect_line_ind.append(i)
+            ind_check = 0
+            for k in range(len(v_inspect)-1):
+                if v_inspect_line_ind[k] == v_inspect_line_ind[k+1]:
+                    np.testing.assert_allclose(pipe_hp[k] * pipe_mass[k+1] / pipe_mass[k], pipe_hp[k+1])
+                elif v_inspect_line_ind[k] < v_inspect_line_ind[k+1]:
+                    np.testing.assert_array_less((pipe_hp[k] + 1e-6)* pipe_mass[k + 1] / pipe_mass[k],
+                                               pipe_hp[k + 1])
+                    ind_check += 1
+                elif v_inspect_line_ind[k] > v_inspect_line_ind[k + 1]:
```

Comment:
I would suggest we make the test such that the flow always increases over time avoiding the last elif statement, and instead giving an error if this is not the case

Reply:
done